src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-details/rgw-bucket-details.component.ts

```diff
@@ -11,15 +13,53 @@ export class RgwBucketDetailsComponent implements OnChanges {
   @Input()
   selection: any;
 
+  aclPermissions: Record<string, string[]> = {};
+
   constructor(private rgwBucketService: RgwBucketService) {}
 
   ngOnChanges() {
     if (this.selection) {
       this.rgwBucketService.get(this.selection.bid).subscribe((bucket: object) => {
         bucket['lock_retention_period_days'] = this.rgwBucketService.getLockDays(bucket);
         this.selection = bucket;
-        this.selection.bucket_policy = JSON.parse(this.selection.bucket_policy) || {};
+        this.aclPermissions = this.parseXmlAcl(this.selection.acl, this.selection.owner);
       });
     }
   }
+
+  parseXmlAcl(xml: any, bucketOwner: string): Record<string, string[]> {
+    const parser = new xml2js.Parser({ explicitArray: false, trim: true });
+    let data: Record<string, string[]> = {
+      Owner: ['-'],
+      AllUsers: ['-'],
+      AuthenticatedUsers: ['-']
+    };
+    parser.parseString(xml, (err, result) => {
+      if (err) {
+        return null;
+      }
+      const xmlGrantees: any = result['AccessControlPolicy']['AccessControlList']['Grant'];
+      if (Array.isArray(xmlGrantees)) {
+        for (let i = 0; i < xmlGrantees.length; i++) {
+          const grantee = xmlGrantees[i];
+          if (grantee.Grantee.URI) {
```

Comment:
you might want to do this , in order to be safe from undefined exceptions
```suggestion
          if (grantee?.Grantee?.URI) {
```

Reply:
same for below instances

Reply:
added them